Automattic/wp-calypso
# Include Boost in product removal confirmation dialog

#### Proposed Changes
* While cancelling a Jetpack Boost plan or Jetpack Complete, warn user that they will miss out on Boost features

![image](https://user-images.githubusercontent.com/3737780/211252087-aab122de-fe52-418a-bb98-766aa31bbb43.png)

![image](https://user-images.githubusercontent.com/3737780/211251869-f820e48d-a177-4594-8041-4385f41763b3.png)

#### Testing Instructions



* Purchase Boost premium on a test site
* While running calypso locally access your purchases http://calypso.localhost:3000/me/purchases
* Remove the plan
* The removal dialog should mention Boost is being removed
* Purchase Jetpack Complete
* Remove the plan
* The removal dialog should mention Boost is being removed

#### Pre-merge Checklist





Related to Automattic/boost-cloud#158
